### DISCUSSION TITLE
Thanks for this!  I have a few questions for the person/ team developing FaceSwapLab

### DISCUSSION THREAD

**Discussion by TrickleDownClown** [glucauze/sd-webui-faceswaplab#87] [Q&A]
I just wanted to take a moment to thank you for creating such an amazing piece of software. Face Swap Lab is incredibly impressive,

I don't have any coding experience, but I would still like to contribute to Face Swap in some way. Is there any way that I can donate to the project? Let me know.  

I also have a few questions about Face Swap.

1. Do you plan on implementing a clip-type feature similar to Rope and Roop Unleashed? This would allow users to swap faces in videos, which would be a really cool feature.
2.  When extracting faces to build a model from, I noticed a number of the images had weird artifacts (namely the eyes) after the extraction. Should I clean these out before building the model? Does it matter? Why does this happen? Is there anything I can do in terms of image selection to avoid this issue? 
3. Are there any best practices for image selection (Size, Resolution etc.) ? 
4. When building the model is it better to have multiple orientations, expressions etc.?  If orientations are accounted for should I flip some of the source images as you would doing LoRA training?
5. One last question.  Is there a limit to how many images/faces you can import to build from?

Thanks again for creating Face Swap. It's amazing, and I'm really looking forward to seeing what you do with it in the future.

Best regards,

**Comment by glucauze** [comment]
Oh thx,  i don't need money right now. I needed the extension to restore some very old photos. I thought it could be useful to make the work public. But you can contribute to doc and share you experiments (no NSFW please) if you want.

1. No i don't plan on doing that. I think https://github.com/RichardErkhov/FastFaceSwap or https://github.com/facefusion/facefusion are doing on a great job on that aspect. 
2. When building a model i display a preview that is probably not representative of the quality of the model itself. I choose Lancszos as upscaler, and codeformer. It is just a preview. So test it on real image before drawing any conclusion on that. 
3. That is to be determine. I did not notice any improvement by using codeformer before but that may help in some case. Don't forget that the resulting model will be 128x128 so i don't think it matters that much.
4. No. I often use 3 or 4 images and see if the result is better. That is what makes insightface so interesting. It doesn't require that much images. But you could test with more if you want.
5. No limit in the code. But there might be one if you use thousands of them.

Thx for your feedback
